### PR TITLE
added question on HTTP and some refactorings

### PR DESCRIPTION
**Pull request by mickaelandrieu**

### CODE REVIEW

File: data/controllers.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ questions:
             - {value: return $this->getContainer()->get('service.name'), correct: false}
             - {value: return $this->container->get('service.name'),      correct: true}
     -
-        question: 'Which class must be extended by your controllers?'
+        question: 'Which class may be extended by your controllers?'
         answers:
             - {value: Symfony\Bundle\FrameworkBundle\Controller,    correct: true}
```
Comment (xabbuh) [inline data/controllers.yml]: This is not true. The FQCN is `Symfony\Bundle\FrameworkBundle\Controller\Controller`.

File: data/http.yml [hunk 2]
```diff
@@ -31,10 +24,50 @@ questions:
...
+        answers:
+            - {value: "$request->query->get('foo');",     correct: true}
+            - {value: "$request->request->get('foo');",   correct: false}
+            - {value: "{$request->query->all()}['foo'];", correct: true}
+            - {value: "$request->get('foo');",            correct: false}
+    -
+        question: 'How to access the `bar` POST parameter from Request object $request ?'
+        answers:
+            - {value: "$request->query->get('bar');",     correct: false}
+            - {value: "$request->request->get('bar');",   correct: false}
```
Comment (xabbuh) [inline data/http.yml]: This answer should be marked as correct.

File: data/http.yml [hunk 3]
```diff
@@ -31,10 +24,50 @@ questions:
...
+            - {value: $request->isXmlHttpRequest();,                                   correct: true}
+            - {value: $request->isAJAX();,                                             correct: false}
+            - {value: "$this->headers->get('X-Requested-With') === 'XMLHttpRequest';", correct: true}
+    -
+        question: 'Theses Response subclasses are available: true or false ?'
+        answers:
+            - {value: JsonResponse,       correct: true}
+            - {value: StreamResponse,     correct: false}
+            - {value: BinaryFileResponse, correct: true}
+            - {value: XmlResponse,        correct: true}
```
Comment (xabbuh) [inline data/http.yml]: This is not true.

Comment (mickaelandrieu) [inline data/http.yml]: :)  to be honest I have created theses "issues"  to check if someone realy use certificationy! :D very honored to have your review \o/

File: data/http.yml (hunk 2 above)
Comment (mickaelandrieu) [inline data/http.yml]: was a real issue,  my bad :/

**Comment by mickaelandrieu** [comment]
anyway, thanks for review @xabbuh !

**Pull request merged**